File: Python Solutions/278. First Bad Version.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return an integer
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        min = 0
        max = n
        previous = None
        while True:
            if isBadVersion(n):
                max = n
            else:
                min = n
            n = int((min + max) / 2)
            if n == previous: return n + 1
            previous = n

File: C Solutions/278. First Bad Version.c
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

int firstBadVersion(int n) {
    long min = 0;
    long max = n;
    int mid;
    while(min <= max){
        mid = (min + max) / 2;
        if(isBadVersion(mid)) max = mid - 1;
        else min = mid + 1;
    } 
    return min;
}
